### DISCUSSION TITLE
Referenced package cannot be installed (python-pip)

### DISCUSSION THREAD

**Discussion by barrct** [dlemstra/Magick.Native#13] [General]
Trying to install this package in the install dep file causes the install to fail and not install all of the required dependances.

python3-pip works fine, no issues there.

uname -a
Linux <redacted> 5.10.0-9-amd64 #1 SMP Debian 5.10.70-1 (2021-09-30) x86_64 GNU/Linux

https://github.com/dlemstra/Magick.Native/blob/d0642c2c3e395fd856a49fef4ba8a25099883098/build/linux/install.dependencies.sh#L6

apt-get install python-pip
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package python-pip is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  python3-pip

E: Package 'python-pip' has no installation candidate

**Comment by dlemstra** [comment]
This dependencies are only required when you want to build the library yourself. If you want to do this yourself I would advise you to build this inside a container. Those dependencies are for `ubuntu:16.04` so this will probably not work for Debian.

**Comment by barrct** [comment]
Wow.
Okay.
So, just to begin with.... This is STILL a depreciated package in Ubuntu, and I even spun up an Ubuntu workstation to verify 100% for you.

Ubuntu:~# sudo ./Magick.Native/build/linux/install.dependencies.sh
Hit:1 http://us.archive.ubuntu.com/ubuntu impish InRelease
Get:2 http://us.archive.ubuntu.com/ubuntu impish-updates InRelease [90.7 kB]
Hit:3 http://us.archive.ubuntu.com/ubuntu impish-backports InRelease    
Get:4 http://security.ubuntu.com/ubuntu impish-security InRelease [90.7 kB]
Fetched 181 kB in 1s (287 kB/s)     
Reading package lists... Done
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'python-is-python2' instead of 'python'
Package python-pip is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  python3-pip

E: Package 'python-pip' has no installation candidate


Ubuntu:~# uname -a
Linux <redacted> 5.13.0-20-generic #20-Ubuntu SMP Fri Oct 15 14:21:35 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux


Ubuntu:~# apt-get install python-pip
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package python-pip is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  python3-pip

E: Package 'python-pip' has no installation candidate

This still means that this is a valid issue with the code that is in the repository.

Secondly, per your instructions on https://github.com/dlemstra/Magick.NET/issues/380#issuecomment-944941384 I need to "clone this repository: https://github.com/dlemstra/Magick.Native. There are no instruction on how to build it with ARM support but you could probably follow the build steps on Linux."
So I did as you directed.
Step 1 is to run "build/linux/install.dependencies.sh"
Step 1 fails because of the packages listed to be installed are invalid.

Since I saw the issue was with an old package that was already updated in the apt install script to python3-pip, I though that I should report the error that breaks the process and causes the build to fail.

Thanks for telling me that I'm stupid though and closing the issue without even looking at anything.

All I'm trying to do is to compile your work on a great tool for an ARM processor like so many others that have Apples and Raspis.
I'm trying to make sure that I can compile the code at all first before I try to make things non-standard and try to compile for an ARM by either compiling it ON the Raspberry Pi or cross compiling on my main workstation.
I thought I was helping by nothing the error, but I guess the code is your baby and you don't want feedback so for that, I am sorry.

**Comment by dlemstra** [comment]
I have read my response multiple times now but I don't see me calling you stupid and this is also something that I would never do. I did not close your issue but I changed it into a discussion because that feels like a better place to have this discussion.

As I said in my response I am building this library on Linux with version `16.04` of Ubuntu inside a container. On that version the `python` and `python-pip` packages are not yet deprecated. Once that version is no longer supported I will probably switch to `18.04` and that also seems to support `python-pip`. I does seem that I will need to take a look at that install script once we reach April 2028 and I will need to switch to version `20.04` where the package is deprecated.
